app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/SpeechFragment.java

```diff
@@ -118,37 +120,30 @@ public void onDetach() {
         mListener = null;
     }
 
-    // TODO: CHANGE THIS WHEN WE HAVE A FANCY LISTENING INDICATOR
-    private void updateListeningIndicator() {
-        if (isListening)
-            listeningIndicator.setText(R.string.listening_text_indicator);
-        else
-            listeningIndicator.setText(R.string.not_listening_text_indicator);
-    }
-
     public void doListen() {
         Log.d(DEBUG_TAG, "Start listening");
         isListening = true;
-        updateListeningIndicator();
+        listenButton.setImageResource(R.drawable.yes_mic);
         sr.startListening(recognizerIntent);
     }
 
     public void dontListen() {
         Log.d(DEBUG_TAG, "Stop listening.");
         sr.stopListening();
+        listenButton.setImageResource(R.drawable.no_mic);
```

Comment:
I don't think it's very necessary. Lucy's code just uses a callback to do it, and otherwise, it's adding more steps to do something that only requires one line. You'd have
```java
updateListeningIndicator(){
if(isListen==true){
setButton(yes_mic)
}
else{
setButton(no_mic)
}
}```

and whenever you need to call `updateListeningIndicator()` you could have just put in `setButtonImage(yes_mic)` or the opposite instead. You're always declaring `isListen=false` or `isListen=true` right before you call `updateListeningIndicator()` anyway, so it feels unnecessary to me.